File: app/src/main/java/dog/snow/androidrecruittest/MainActivity.java
package dog.snow.androidrecruittest;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

import dog.snow.androidrecruittest.data_repository.SnowDogViewModel;
import dog.snow.androidrecruittest.ui.list.SnowDogListFragment;

public class MainActivity extends AppCompatActivity {


    private SnowDogViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try
        {
            this.getSupportActionBar().hide();
        }
        catch (NullPointerException e){}
        setContentView(R.layout.activity_main);
        viewModel = ViewModelProviders.of(this).get(SnowDogViewModel.class);
//        SnowDogDatabase.Companion.getInstance(this);
        getSupportFragmentManager().beginTransaction().replace(R.id.conteiner_fragment,new SnowDogListFragment()).commit();
    }
}

File: app/src/main/java/dog/snow/androidrecruittest/ui/page/main/MainActivity.kt
package dog.snow.androidrecruittest.ui.page.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.ui.setupActionBarWithNavController
import dagger.android.AndroidInjection
import dog.snow.androidrecruittest.R
import dog.snow.androidrecruittest.databinding.ActivityMainBinding
import kotlinx.android.synthetic.main.layout_toolbar.view.*

class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        setSupportActionBar( binding.layoutAppbar.appbar.toolbar)
        navController = findNavController(R.id.nav_host_fragment)
        setupActionBarWithNavController(navController)
    }

    override fun onSupportNavigateUp(): Boolean {
        navController.navigateUp()
        return true
    }

}